allennlp/models/decomposable_attention.py

```diff
@@ -164,13 +165,29 @@ def forward(self,  # type: ignore
 
         output_dict = {"label_logits": label_logits, "label_probs": label_probs}
 
+        self.add_insight_to_output_dict(metadata, output_dict, p2h_attention, h2p_attention)
```

Comment:
After switching the `MetadataField` to something on the `Predictor`, it seems a bit of overkill to add a method just to add the two attention fields to `output_dict`.  I'd just put those two lines inline here.